pkg/controller/association/ca.go

```diff
@@ -53,12 +47,12 @@ func ReconcileCASecret(
 	expectedSecret := corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: associated.GetNamespace(),
-			Name:      ElasticsearchCACertSecretName(associated, suffix),
-			Labels:    labels,
+			Name:      ElasticsearchCACertSecretName(associated, r.AssociationName),
+			Labels:    r.AssociationLabels(k8s.ExtractNamespacedName(associated)),
```

Comment:
we used to set the association labels + the associated resource labels here:
```yaml
"labels": {
            "apm.k8s.elastic.co/name": "kibana-sample",
            "common.k8s.elastic.co/type": "apm-server",
            "kibanaassociation.k8s.elastic.co/name": "kibana-sample",
            "kibanaassociation.k8s.elastic.co/namespace": "default"
        },
```

I think this is not really necessary, and other secrets created for an association don't have this.
I changed it to having only the association labels:

```yaml
"labels": {
            "kibanaassociation.k8s.elastic.co/name": "kibana-sample",
            "kibanaassociation.k8s.elastic.co/namespace": "default"
        },
```